### PR TITLE
[MINOR][DOCS] Document 'without' value for HADOOP_VERSION in pip installation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

I believe it's self-descriptive.

### Why are the changes needed?

To document supported features.

### Does this PR introduce _any_ user-facing change?

Yes, the docs are updated. It's master only.

### How was this patch tested?

Manually built the docs via `cd python/docs` and `make clean html`:

Screen Shot 2020-11-20 at 10 59 07 AM

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @ueshin can you take a quick look please?

File: python/docs/source/getting_started/install.rst [hunk 1]
```diff
@@ -68,8 +68,8 @@ It is recommended to use ``-v`` option in ``pip`` to track the installation and
 
     HADOOP_VERSION=2.7 pip install pyspark -v
 
-Supported versions of Hadoop are ``HADOOP_VERSION=2.7`` and ``HADOOP_VERSION=3.2`` (default).
-Note that this installation of PySpark with a different version of Hadoop is experimental. It can change or be removed between minor releases.
+Supported values in ``HADOOP_VERSION`` are ``without``, ``2.7`` and ``3.2`` (default).
```
Comment (ueshin) [inline python/docs/source/getting_started/install.rst:71]: I might want to know what `without` means, like it's equivalent to `user-provided` one or something.

**Review by ueshin — COMMENTED** [review]
Otherwise, LGTM.

File: python/docs/source/getting_started/install.rst (hunk 1 above)
Comment (HyukjinKwon) [reply python/docs/source/getting_started/install.rst:71]: Sure, let me update.

File: python/docs/source/getting_started/install.rst [hunk 2]
```diff
@@ -68,8 +68,13 @@ It is recommended to use ``-v`` option in ``pip`` to track the installation and
 
     HADOOP_VERSION=2.7 pip install pyspark -v
 
-Supported versions of Hadoop are ``HADOOP_VERSION=2.7`` and ``HADOOP_VERSION=3.2`` (default).
-Note that this installation of PySpark with a different version of Hadoop is experimental. It can change or be removed between minor releases.
+Supported values in ``HADOOP_VERSION`` are:
+
+- ``without``: Spark pre-built with user-provided Apache Hadoop
+- ``2.7``: Spark pre-built for Apache Hadoop 2.7
+- ``3.2``: Spark pre-built for Apache Hadoop 3.2 and later (default)
```
Comment (ueshin) [inline python/docs/source/getting_started/install.rst:75]: Awesome!

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**